examples/move/big_set/sources/big_set.move

```diff
@@ -0,0 +1,215 @@
+// Copyright (c) Mysten Labs, Inc.
+// SPDX-License-Identifier: Apache-2.0
+
+/// A BigSet is a VecSet-like which can go beyond object limit by utilizing
+/// dynamic fields. Either by 256 or 65536 times depending on the scaling.
+///
+/// The structure of this module is similar to the VecSet module, but with some
+/// differences:
+/// - BigSet requires TxContext due to UID requirement
+/// - BigSet can not have `copy` ability, and there's a `copy_` function instead
+/// - BigSet can not have `drop` ability, and there's a `drop` function
+/// - Returning references to stored values doesn't seem to be possible atm
+///
+/// The storage structure is as follows:
+///
+/// 1. When a value is inserted, we hash it and take the first byte (or two)
+/// to get the key for the dynamic field.
+/// 2. Then we create or add to the VecSet stored in the dynamic field.
+/// 3. We also add the key to the VecSet of keys to keep track of all the
+/// dynamic fields (for `copy_`, `drop` and `into_keys`).
+module big_set::big_set {
+    use std::vector;
+
+    use sui::vec_set::{Self, VecSet};
+    use sui::tx_context::TxContext;
+    use sui::object::{Self, UID};
+    use sui::dynamic_field as df;
+    use sui::hash::blake2b256;
+    use sui::bcs;
+
+    /// Error code for unimplemented functions.
+    const ENotImplemented: u64 = 0;
+    #[allow(unused_const)]
+    /// Error code for invalid scaling (currently only 1 and 2 are supported).
+    const EInvalidScaling: u64 = 1;
+    /// Error code for a key that does not exist in the set.
+    const EKeyDoesNotExist: u64 = 2;
+
+    /// An item Key; used to store items in the set.
+    struct Key has copy, store, drop { value: vector<u8> }
```

Comment:
a vector here seems an overkill given the implementation. I would go for a `u16` but not a big deal anyway.
Of course I may be missing something behind the intent here